ns/list.n3

```diff
@@ -0,0 +1,112 @@
+#   n3 definition of some list functions
+#
+# $Id: list.n3,v 1.3 2010-03-26 20:41:54 timbl Exp $
+
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix doc: <http://www.w3.org/2000/01/pim/doc#>.
+@prefix log: <http://www.w3.org/2000/10/swap/log#>.
+@prefix owl: <http://www.w3.org/2002/07/owl#>.
+@prefix vs: <http://www.w3.org/2003/06/sw-vocab-status/ns#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+@prefix list: <http://www.w3.org/2000/10/swap/list#>.
+@prefix : <#>.
+
+list:append a rdf:Property, owl:FunctionalProperty;
+    rdfs:label "append";
+    rdfs:domain log:List;
+    rdfs:comment """
+Iff the subject is a list of lists and the concatenation of all those lists 
+is the object, then this is true.
+eg  ( (1 2) (3 4) ) list:append (1 2 3 4).
+The object can be calculated as a function of the subject.
+""";
+    rdfs:isDefinedBy <>.
+
+list:first a rdf:Property, owl:FunctionalProperty;
+    rdfs:label "first";
+    rdfs:domain log:List;
+    rdfs:comment """
+Iff the subject is a list and the object is the first thing that list, then this is true.
+The object can be calculated as a function of the list.
+""";
+    vs:term_status "unstable";
+    rdfs:isDefinedBy <>.
+
+list:in a rdf:Property;
+    rdfs:label "in";
+    rdfs:range log:List;
+    rdfs:comment """
+Iff the object is a list and the subject is in that list, then this is true.
+""";
+    rdfs:isDefinedBy <>.
+
+list:last a rdf:Property, owl:FunctionalProperty;
+    rdfs:label "in";
+    rdfs:domain log:List;
+    rdfs:comment """
+Iff the subject is a list and the object is the last thing that list, then this is true.
+The object can be calculated as a function of the list.
+""";
+    rdfs:isDefinedBy <>.
+
+list:length a rdf:Property, owl:FunctionalProperty;
+    rdfs:label "length";
+    rdfs:domain log:List;
+    rdfs:range xsd:integer;
+    rdfs:comment """
+Iff the subject is a list and the object is the last thing that list, then this is true.
+The object can be calculated as a function of the list.
+""";
+    vs:term_status "unstable";
+    rdfs:isDefinedBy <>.
+
+list:member a rdf:Property;
+    rdfs:label "member";
+    rdfs:domain log:List;
+    rdfs:comment """
+Iff the subject is a list and the object is in that list, then this is true.
+""";
+    rdfs:isDefinedBy <>.
+
+list:memberAt a rdf:Property;
+    rdfs:label "memberAt";
+    rdfs:domain log:List;
+    rdfs:comment """
```

Comment:
Updated from the builtin formalization doc. We could consider using Markdown, and we might split among some different annotation properties.